src/experimental/constructs/iam/roles/cross-account-role.ts

```diff
@@ -0,0 +1,42 @@
+import { ArnPrincipal } from "aws-cdk-lib/aws-iam";
+import type { GuStack } from "../../../../constructs/core";
+import type { GuRoleProps} from "../../../../constructs/iam";
+import {GuRole} from "../../../../constructs/iam";
+
+export interface GuCrossAccountRoleExperimentalProps extends Omit<GuRoleProps, "assumedBy"> {
+  /**
+   * The name of the role that will assume this new GuCrossAccountRoleExperimental.
+   * It should be the full name including the stage (if appropriate).
+   */
+  nameOfRoleWhichCanAssumeThisRole: string;
+
+  /**
+   * The AWS account ID associated with the resource that will consume this new GuCrossAccountRoleExperimental.
+   *
+   * If this construct is being used on a public repo it would be advisable to use
+   * [private-infrastructure-config](https://github.com/guardian/private-infrastructure-config) so that any account ID
+   * is not publicly available.
+   */
+  accountId: string;
+}
+
+/**
```

Comment:
This is the correct place to add the docs, thanks! 

It looks like there may be something wrong with the [documentation setup](https://github.com/guardian/cdk/blob/main/typedoc.js), because [I can't see the docs for some of the other experimental constructs either](https://guardian.github.io/cdk/modules/experimental_constructs.html).

I think we might need to update https://github.com/guardian/cdk/blob/main/src/experimental/constructs/index.ts and/or https://github.com/guardian/cdk/blob/main/src/experimental/constructs/iam/index.ts to fix this - perhaps one for another PR unless it's simple!

Reply:
Thanks for pointing this out! I think it was relatively simple fix, I added the exports to constructs/index.ts and could then see the constructs when generating the typedocs locally:

<img width="1402" alt="Screenshot 2023-10-11 at 12 23 27" src="https://github.com/guardian/cdk/assets/45561419/5bbbcbae-949c-4bde-9a23-c4590cb42dec">